### ISSUE TITLE
iOS 1.68.0 lost dns with exit node enabled.. 

### ISSUE THREAD

**Issue by bcookatpcsd** [tailscale/tailscale#12475]
### What is the issue?

tl:dr; disable use as exit node to get dns working.. 

tailscale as road warrior setup.. just connecting back to the residence when on mobile data.. 

iOS, 17.4.1 messing with dns on a machine 100.75.66.53.. recently the iphones stopped showing up in the logs.. toggled, changed, etc, in tailscale.. using the same machine as an exit node.. 

dns leak test showed some res730.ord.rrdns.pch.net, res761, and res762 as the recursors.. when on mobile data with verizon.. after doing anything and everything to try and see my dns data appear in tailscale.. I had to *log out* and *log back in* to my tailnet to get the dns settings to work with my server settings.. 

Signing out does disable the exit node.. which is the odd part.. with the exit node disabled I would see all dns.. with exit node enabled.. it seems I lose dns.. 

there was no way to tell it wasn't working with my settings.. 

vpn was connected all tailscale settings looked enabled

Seems to be pretty reproduceable.. 1.68.0

Anyone else? 

### Steps to reproduce

above.. 

### OS

iOS

### OS version

17.4.1

### Tailscale version

1.68.0

### Bug report

BUG-8bd80029c5bac9223fe518d4d4022b1e4c8fbde81af55a30881f8a6a23eb7ad5-20240614202337Z-b256d704e779c48e

**Comment by agottardo** [comment]
Do you have "Use Tailscale DNS Settings" enabled or disabled in the app? (Asking because it determines what kind of configuration we install).

**Comment by bcookatpcsd** [comment]
![image](https://github.com/tailscale/tailscale/assets/55087301/7fcfa274-498a-4007-b826-bc5510a84573)

![image](https://github.com/tailscale/tailscale/assets/55087301/e14ffe93-0821-4dc0-bd9c-47f19b7ffebc)

Again.. exit node is 66.53 and when you tail the logs on the resolver (smartdns or blocky) nothing shows until you disable the exit node..

Please let me know if you need/want anything else..

**Comment by agottardo** [comment]
Can you provide a bug report ID taken on your iPhone? The one you sent is from the exit node.

**Comment by bcookatpcsd** [comment]
Exit node enabled:

BUG-d88ca3d8b83c69a510217e97dae0a683369b92376c3a5789b06b55680c979cd4-20240615163111Z-8bd9b109f026de18

Exit node disabled:

BUG-d88ca3d8b83c69a510217e97dae0a683369b92376c3a5789b06b55680c979cd4-20240615163140Z-86467eaa7b52ee8f

**Comment by agottardo** [comment]
Metrics indicate your iPhone is sending DNS queries to 100.75.66.53 over the exit node as you expected, but such DNS lookups are failing because that server is not responding:

```
2024-06-15 14:32:58	dns udp query: waiting for response or error from [100.75.66.53]: context deadline exceeded
[...]
2024-06-15 14:35:02	dns: resolver: forward: recv: response code indicating server failure: 2
[...]
2024-06-15 14:35:07	dns udp query: resolving using "100.75.66.53": response code indicates server issue dial tcp4 100.75.66.53:53: i/o timeout
```

You can look at these logs yourself by looking for log entries from the IPNExtension process when your iPhone is connected to a Mac, in the Console app.

Just to confirm, is the exit node allowed to talk to 100.75.66.53? Does 100.75.66.53 support DNS over HTTPS, or only regular DNS queries over UDP port 53?

**Comment by bcookatpcsd** [comment]
And when the exit node is disabled the server works..

So the server is working when the exit node is disabled..

**Comment by bcookatpcsd** [comment]
And the same thing happens with Quad9 and Cloudflare.

Enable the exit node lose the tailnet dns.

**Comment by bcookatpcsd** [comment]
![image](https://github.com/tailscale/tailscale/assets/55087301/44c8d9b6-8859-4b38-9a69-be0c37a007b0)

![image](https://github.com/tailscale/tailscale/assets/55087301/1ebad186-91ea-4ca4-a454-514e935db020)

![image](https://github.com/tailscale/tailscale/assets/55087301/4ece0088-1248-480e-b17d-5b8125bac41c)

**Comment by bcookatpcsd** [comment]
![image](https://github.com/tailscale/tailscale/assets/55087301/0fa19562-a0c3-4541-9115-7faed10490b9)

![image](https://github.com/tailscale/tailscale/assets/55087301/11288f47-b682-4688-bc19-66312cba5edf)

![image](https://github.com/tailscale/tailscale/assets/55087301/60f7cc3c-0948-4093-88d9-63fc5c74a55f)

If my dns and Quad9 were both down.. where did the (assuming) mobile provider dns come from..

**Comment by bcookatpcsd** [comment]
#12467..
